### ISSUE TITLE
distinct types with static[T] can't borrow

### ISSUE THREAD

**Issue by cooldome** [nim-lang/Nim#6026]
The following code snippet does not compile, while there is no particular reason not to.

```
type Amount[date: static[int]] = distinct float

proc `+`[T](x, y: Amount[T]): Amount[T] {. borrow .}
```

**Comment by andreaferretti** [comment]
Is this the same as https://github.com/nim-lang/Nim/issues/4121 ?

**Comment by cooldome** [comment]
Yeah looks very close, however static[T] will likely need extra treatment

**Comment by bung87** [comment]
this one problem is not related to static but magic, I see comments in https://github.com/nim-lang/Nim/pull/6232, so the magic borrow in not including in design.
